genai-perf/genai_perf/inputs/output_format_converter.py

```diff
@@ -0,0 +1,780 @@
+        user_role_headers: List[str],
+        openai_json: Dict,
+    ) -> None:
+        if "image" in entry:
+            contents = self._extract_chat_contents(entry)
+            if openai_json["payload"][0]["messages"]:
+                openai_json["payload"][0]["messages"][0]["content"].extend(contents)
+            else:
+                openai_json["payload"][0]["messages"].append(
+                    {"role": "user", "content": contents}
+                )
+        else:
+            for header, content in entry.items():
+                message = self._create_new_openai_chat_completions_message(
+                    header, system_role_headers, user_role_headers, content
+                )
+                self._add_message_to_json(openai_json, message)
+
+    def _extract_chat_contents(self, entry: Dict) -> List[Dict]:
+        contents: List = []
+        if isinstance(entry, list):
+            for item in entry:
+                for content_type, content in item.items():
+                    self._add_content(contents, content_type, content)
+        else:
+            for content_type, content in entry.items():
+                self._add_content(contents, content_type, content)
+        return contents
+
+    def _add_content(self, contents: List[Dict], content_type: str, content: str):
+        if content_type == "text_input":
+            contents.append({"type": "text", "text": content})
+        elif content_type == "image":
+            contents.append({"type": "image_url", "image_url": {"url": content}})
+        else:
+            raise GenAIPerfException(
+                "Failed to construct OpenAI chat completions message "
+                f"contents. Unknown content type: '{content_type}'."
+            )
+
+    def _create_new_openai_chat_completions_message(
+        self,
+        header: str,
+        system_role_headers: List[str],
+        user_role_headers: List[str],
+        content: str,
+    ) -> Optional[Dict]:
+        # Do not add messages with blank content
+        if not content:
+            return {}
+
+        if header in system_role_headers:
+            message = {
+                "role": "system",
+                "content": content,
+            }
+        elif header in user_role_headers:
+            message = {
+                "role": "user",
+                "content": content,
+            }
+        else:
+            message = {}
+
+        return message
+
+    def _create_new_prompt(
+        self,
+        header: str,
+        system_role_headers: List[str],
+        user_role_headers: List[str],
+        text_input_headers: List[str],
+        content: str,
+    ) -> str:
+        new_prompt = ""
+
+        if (
+            header in system_role_headers
+            or header in user_role_headers
+            or header in text_input_headers
+        ):
+            new_prompt = content
+
+        return new_prompt
+
+    def _create_new_text_input(
```

Comment:
If this class is being derived from BaseConverter and the methods are identical why aren't we just using the base converter class' method?
Also, it seems like _create_new_prompt and _create_new_text_input are identical. Maybe it would be better to create a generic method in BaseConverter and then just call this for both methods?

Reply:
Thanks for catching this! I fixed this. I was trying to use the lightest touch when pulling these out of the original file and missed this duplication.